beets/util/artresizer.py

```diff
@@ -232,23 +247,34 @@ def _check_method():
 
 def get_im_version():
     """Return Image Magick version or None if it is unavailable
-    Try invoking ImageMagick's "convert".
+    Try invoking ImageMagick's "magick". If "magick" is unavailable,
+    as with older versions, fall back to "convert"
+
+    Our iterator will be non-zero when the first command fails, and will
+    be returned in a tuple along with the version.
     """
-    try:
-        out = util.command_output(['convert', '--version'])
-
-        if b'imagemagick' in out.lower():
-            pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
-            match = re.search(pattern, out)
-            if match:
-                return (int(match.group(1)),
-                        int(match.group(2)),
-                        int(match.group(3)))
-            return (0,)
-
-    except (subprocess.CalledProcessError, OSError) as exc:
-        log.debug(u'ImageMagick check `convert --version` failed: {}', exc)
-        return None
+    cmd_names = (['magick'],
+                 ['convert'])
+    for i, cmd_name in enumerate(cmd_names):
```

Comment:
I recommend something a little more obvious than using `enumerate`: something like this.

```py
for legacy, cmd_name in ((False, 'magick'), (True, 'convert')):
```

This avoids the need for a `bool(i)` conversion later on.

Reply:
The bool conversion is actually unnecessary, since we really only need to evaluate `i` as non-zero, but I thought the intended purpose would be more clear as a bool. I added an explanation to the docstring on line 257 which I thought was reasonably clear, but your approach is perhaps more concise anyway.

Reply:
The bool conversion is actually unnecessary, since we really only need to evaluate `i` as non-zero, but I thought the intended purpose would be more clear as a bool. I added an explanation to the docstring on line 257 which I thought was reasonably clear, but your approach is perhaps more concise anyway.